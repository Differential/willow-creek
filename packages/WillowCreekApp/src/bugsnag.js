import { Client } from 'bugsnag-react-native';
import Config from 'react-native-config';

const bugsnag = new Client(Config.BUGSNAG_API_KEY);

export default bugsnag;
