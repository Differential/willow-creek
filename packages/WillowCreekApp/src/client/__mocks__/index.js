import { MockedProvider } from 'react-apollo/test-utils';
import ApolloClient from 'apollo-client';
import { makeExecutableSchema, addMockFunctionsToSchema } from 'graphql-tools';
import { SchemaLink } from 'apollo-link-schema';
import { ApolloLink } from 'apollo-link';
import { testSchema as typeDefs } from 'apollos-church-api';

import cache from '../cache';
import clientStateLink from '../clientStateLink';

export default MockedProvider;

const schema = makeExecutableSchema({
  typeDefs,
  resolverValidationOptions: {
    requireResolversForResolveType: false,
  },
});
addMockFunctionsToSchema({ schema });

let link = new SchemaLink({ schema });
if (clientStateLink) {
  link = ApolloLink.from([clientStateLink, link]);
}

export const client = new ApolloClient({
  cache,
  link,
});
