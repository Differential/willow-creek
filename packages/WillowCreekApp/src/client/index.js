import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { ApolloProvider } from 'react-apollo';
import { ApolloClient } from 'apollo-client';
import { ApolloLink } from 'apollo-link';
import SplashScreen from 'react-native-splash-screen';

import httpLink from './httpLink';
import clientStateLink from './clientStateLink';
import authLink from './authLink'; // eslint-disable-line
import cache, { ensureCacheHydration } from './cache';

const link = clientStateLink.concat(ApolloLink.from([authLink, httpLink]));

export const client = new ApolloClient({
  link,
  cache,
  queryDeduplication: true,
  shouldBatch: true,
});

class ClientProvider extends PureComponent {
  static propTypes = {
    client: PropTypes.shape({
      cache: PropTypes.shape({}),
    }),
  };

  static defaultProps = {
    client,
  };

  async componentDidMount() {
    try {
      await ensureCacheHydration;
    } catch (e) {
      throw e;
    } finally {
      if (SplashScreen && SplashScreen.hide) SplashScreen.hide();
    }
  }

  render() {
    return <ApolloProvider {...this.props} client={client} />;
  }
}

export default ClientProvider;
