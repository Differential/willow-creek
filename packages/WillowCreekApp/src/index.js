import React from 'react';
import { StatusBar } from 'react-native';
import { createStackNavigator } from 'react-navigation';
// import { Sentry } from 'react-native-sentry';

import { BackgroundView, withTheme } from '@apollosproject/ui-kit';
import MediaPlayer from 'WillowCreekApp/src/ui/MediaPlayer';
import Providers from './Providers';
import NavigationService from './NavigationService';
import ContentSingle from './content-single';
import Tabs from './tabs';
import Auth from './auth';
import PersonalDetails from './user-settings/PersonalDetails';
import ChangePassword from './user-settings/ChangePassword';

const AppStatusBar = withTheme(({ theme }) => ({
  barStyle: 'dark-content',
  backgroundColor: theme.colors.paper,
}))(StatusBar);

const AppNavigator = createStackNavigator(
  {
    Tabs,
    ContentSingle,
    Auth,
    PersonalDetails,
    ChangePassword,
  },
  {
    initialRouteName: 'Tabs',
    mode: 'modal',
    headerMode: 'screen',
  }
);

const App = () => (
  <Providers>
    <BackgroundView>
      <AppStatusBar barStyle="dark-content" />
      <AppNavigator
        ref={(navigatorRef) => {
          NavigationService.setTopLevelNavigator(navigatorRef);
        }}
      />
      <MediaPlayer />
    </BackgroundView>
  </Providers>
);

export default App;
