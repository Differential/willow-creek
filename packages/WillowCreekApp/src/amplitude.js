import amplitude from 'amplitude-js';
import Config from 'react-native-config';
import gql from 'graphql-tag';
import { get } from 'lodash';
import { getVersion } from 'react-native-device-info';
import { client } from './client';

amplitude.getInstance().init(Config.AMPLITUDE_KEY, null, {
  useNativeDeviceInfo: true,
});

export const track = ({ eventName, properties = null }) => {
  amplitude
    .getInstance()
    .logEvent(eventName, properties, (...args) => console.log(args));
};

export const identify = () => {
  const data = client.readQuery({
    query: gql`
      query currentCampusId {
        currentUser {
          id
          profile {
            id
            campus {
              id
              name
            }
          }
        }
      }
    `,
  });

  amplitude.getInstance().setUserId(get(data, 'currentUser.profile.id'));

  amplitude.getInstance().setUserProperties({
    campusName: get(data, 'currentUser.profile.campus.name'),
    appVersion: getVersion(),
  });
};
