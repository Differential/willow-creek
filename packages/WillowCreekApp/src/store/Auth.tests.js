import { client } from 'WillowCreekApp/src/client';

import { resolvers } from '.';

describe('Auth Store', () => {
  it('logs a user out', async () => {
    client.cache.writeData({ data: { authToken: 'some-auth-token' } });
    resolvers.Mutation.logout({}, {}, { cache: client.cache });
    expect(client.cache).toMatchSnapshot();
  });
});
