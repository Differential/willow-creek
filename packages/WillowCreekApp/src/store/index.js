import { merge, get } from 'lodash';
import gql from 'graphql-tag';
import getLoginState from 'WillowCreekApp/src/auth/getLoginState';
import { track, events } from 'WillowCreekApp/src/analytics';
import { client } from '../client'; // eslint-disable-line
import getAuthToken from './getAuthToken';
// TODO: this will require more organization...ie...not keeping everything in one file.
// But this is simple while our needs our small.

export const schema = `
  type Query {
    authToken: String
    mediaPlayer: MediaPlayerState
    isLoggedIn: Boolean
  }

  type Mutation {
    logout
    mediaPlayerUpdateState(isPlaying: Boolean, isFullscreen: Boolean, isVisible: Boolean): Boolean
    mediaPlayerSetPlayhead(currentTime: Float): Boolean
    mediaPlayerPlayNow(
      parentId: ID,
      mediaSource: VideoMediaSource!,
      posterSources: [ImageMediaSource],
      title: String,
      artist: String,
      isVideo: Boolean,
    ): Boolean

    handleLogin(authToken: String!)
  }

  type MediaPlayerState {
    currentTrack: MediaPlayerTrack
    isPlaying: Boolean
    isFullscreen: Boolean
    isVisible: Boolean
    currentTime: Float
  }

  type MediaPlayerProgress {
    currentTime: Float
    playableDuration: Float
    seekableDuration: Float
    duration: Float
  }

  type MediaPlayerTrack {
    id: ID!
    parentId: ID
    mediaSource: VideoMediaSource!
    posterSources: [ImageMediaSource]
    title: String
    artist: String
    isVideo: Boolean
  }
`;

export const defaults = {
  __typename: 'Query',
  authToken: null,
  mediaPlayer: {
    __typename: 'MediaPlayerState',
    currentTrack: null,
    isPlaying: false,
    isFullscreen: false,
    isVisible: false,
    currentTime: 0,
    showVideo: true,
    muted: false,
  },
};

let trackId = 0;

export const resolvers = {
  Query: {
    isLoggedIn: () => {
      // When logging out, this query returns an error.
      // Rescue the error, and return false.
      try {
        const { authToken } = client.readQuery({ query: getAuthToken });
        return !!authToken;
      } catch (e) {
        return false;
      }
    },
  },
  Mutation: {
    logout: (root, variables, { cache }) => {
      client.resetStore();
      cache.writeData({
        data: defaults,
      });
      track({ eventName: events.UserLogout });
      return null;
    },

    handleLogin: async (root, { authToken }, { cache }) => {
      try {
        await cache.writeQuery({
          query: getAuthToken,
          data: { authToken },
        });
        await cache.writeQuery({
          query: getLoginState,
          data: { isLoggedIn: true },
        });
        await cache.writeData({
          data: { authToken },
        });

        track({ eventName: events.UserLogin });
      } catch (e) {
        throw e.message;
      }

      return null;
    },

    mediaPlayerPlayNow: (root, trackInfo, { cache }) => {
      const query = gql`
        query {
          mediaPlayer {
            isFullscreen
          }
        }
      `;
      const mediaTrack = merge(
        {
          __typename: 'MediaPlayerTrack',
          parentId: null,
          mediaSource: null,
          posterSources: null,
          title: null,
          artist: null,
          isVideo: false,
        },
        trackInfo
      );

      const { mediaPlayer } = cache.readQuery({ query });

      const newMediaPlayerState = {
        __typename: 'MediaPlayerState',
        isPlaying: true,
        isVisible: true,
        isFullscreen: mediaTrack.isVideo
          ? true
          : (mediaPlayer && mediaPlayer.isFullscreen) || false,
        currentTrack: mediaTrack,
        currentTime: 0,
        showVideo: mediaTrack.isVideo,
        muted: false,
      };

      if (
        // if same track
        mediaPlayer &&
        mediaPlayer.currentTrack &&
        mediaPlayer.currentTrack.mediaSource.uri === mediaTrack.mediaSource.uri
      ) {
        // use the same Id
        newMediaPlayerState.currentTrack.id = mediaPlayer.currentTrack.id;
      } else {
        // otherwise reset progress and use new Id
        newMediaPlayerState.currentTrack.id = trackId;
        newMediaPlayerState.progress = null;
        trackId += 1;
      }

      cache.writeData({
        query,
        data: {
          mediaPlayer: newMediaPlayerState,
        },
      });

      track({
        eventName: events.UserPlayedMedia,
        properties: {
          uri: mediaTrack.uri,
          title: mediaTrack.title,
          type: mediaTrack.isVideo ? 'Video' : 'Audio',
        },
      });
      return null;
    },
    mediaPlayerUpdateState: (
      root,
      { isPlaying, isFullscreen, isVisible, showVideo, muted },
      { cache }
    ) => {
      const query = gql`
        query {
          mediaPlayer @client {
            isPlaying
            isFullscreen
            isVisible
            showVideo
            muted
          }
        }
      `;
      const { mediaPlayer } = cache.readQuery({ query });
      cache.writeQuery({
        query,
        data: {
          mediaPlayer: merge(mediaPlayer, {
            isPlaying,
            isFullscreen,
            isVisible,
            showVideo,
            muted,
            __typename: 'MediaPlayerState',
          }),
        },
      });
      return null;
    },
    mediaPlayerSetPlayhead: (root, { currentTime }, { cache }) => {
      const query = gql`
        query {
          mediaPlayer @client {
            currentTime
          }
        }
      `;
      const { mediaPlayer } = cache.readQuery({ query });
      cache.writeQuery({
        query,
        data: {
          mediaPlayer: {
            __typename: 'MediaPlayerState',
            currentTime:
              currentTime || get(mediaPlayer.progress, 'currentTime') || 0,
          },
        },
      });
      return null;
    },
  },
};
