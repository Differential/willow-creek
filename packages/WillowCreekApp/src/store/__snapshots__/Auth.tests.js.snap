// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Store logs a user out 1`] = `
InMemoryCache {
  "addTypename": true,
  "cacheKeyRoot": CacheKeyNode {
    "children": null,
    "key": null,
  },
  "config": Object {
    "addTypename": true,
    "cacheRedirects": Object {
      "Query": Object {
        "node": [Function],
      },
    },
    "dataIdFromObject": [Function],
    "fragmentMatcher": IntrospectionFragmentMatcher {
      "isReady": true,
      "match": [Function],
      "possibleTypesMap": Object {
        "ContentItem": Array [
          "UniversalContentItem",
          "DevotionalContentItem",
        ],
        "Media": Array [
          "ImageMedia",
          "VideoMedia",
          "AudioMedia",
        ],
        "MediaSource": Array [
          "ImageMediaSource",
          "VideoMediaSource",
          "AudioMediaSource",
        ],
        "Node": Array [
          "ContentChannel",
          "Person",
          "Interaction",
          "UniversalContentItem",
          "DevotionalContentItem",
        ],
        "Sharable": Array [
          "SharableContentItem",
        ],
      },
    },
  },
  "data": DepTrackingCache {
    "data": Object {
      "$ROOT_QUERY.mediaPlayer": Object {
        "__typename": "MediaPlayerState",
        "currentTime": 0,
        "currentTrack": null,
        "isFullscreen": false,
        "isPlaying": false,
        "isVisible": false,
        "muted": false,
        "showVideo": true,
      },
      "ROOT_QUERY": Object {
        "__typename": "Query",
        "authToken": null,
        "mediaPlayer": Object {
          "generated": true,
          "id": "$ROOT_QUERY.mediaPlayer",
          "type": "id",
          "typename": "MediaPlayerState",
        },
      },
    },
    "depend": [Function],
  },
  "maybeBroadcastWatch": [Function],
  "optimistic": Array [],
  "silenceBroadcast": false,
  "storeReader": StoreReader {
    "cacheKeyRoot": CacheKeyNode {
      "children": null,
      "key": null,
    },
    "executeSelectionSet": [Function],
    "executeStoreQuery": [Function],
    "keyMaker": QueryKeyMaker {
      "cacheKeyRoot": CacheKeyNode {
        "children": null,
        "key": null,
      },
      "perQueryKeyMakers": Map {},
    },
  },
  "storeWriter": StoreWriter {},
  "typenameDocumentCache": Map {
    Object {
      "definitions": Array [
        Object {
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "GeneratedClientQuery",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "__typename",
                },
              },
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "authToken",
                },
              },
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "mediaPlayer",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "__typename",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "currentTrack",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isPlaying",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isFullscreen",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isVisible",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "currentTime",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "showVideo",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "muted",
                      },
                    },
                  ],
                },
              },
            ],
          },
        },
      ],
      "kind": "Document",
    } => Object {
      "definitions": Array [
        Object {
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "GeneratedClientQuery",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "__typename",
                },
              },
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "authToken",
                },
              },
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "mediaPlayer",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "__typename",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "currentTrack",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isPlaying",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isFullscreen",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isVisible",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "currentTime",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "showVideo",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "muted",
                      },
                    },
                  ],
                },
              },
            ],
          },
        },
      ],
      "kind": "Document",
    },
    Object {
      "definitions": Array [
        Object {
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "GeneratedClientQuery",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "__typename",
                },
              },
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "authToken",
                },
              },
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "mediaPlayer",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "__typename",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "currentTrack",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isPlaying",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isFullscreen",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isVisible",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "currentTime",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "showVideo",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "muted",
                      },
                    },
                  ],
                },
              },
            ],
          },
        },
      ],
      "kind": "Document",
    } => Object {
      "definitions": Array [
        Object {
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "GeneratedClientQuery",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "__typename",
                },
              },
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "authToken",
                },
              },
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "mediaPlayer",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "__typename",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "currentTrack",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isPlaying",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isFullscreen",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isVisible",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "currentTime",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "showVideo",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "muted",
                      },
                    },
                  ],
                },
              },
            ],
          },
        },
      ],
      "kind": "Document",
    },
    Object {
      "definitions": Array [
        Object {
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "GeneratedClientQuery",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "authToken",
                },
              },
            ],
          },
        },
      ],
      "kind": "Document",
    } => Object {
      "definitions": Array [
        Object {
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "GeneratedClientQuery",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "authToken",
                },
              },
            ],
          },
        },
      ],
      "kind": "Document",
    },
    Object {
      "definitions": Array [
        Object {
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "GeneratedClientQuery",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "authToken",
                },
              },
            ],
          },
        },
      ],
      "kind": "Document",
    } => Object {
      "definitions": Array [
        Object {
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "GeneratedClientQuery",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "authToken",
                },
              },
            ],
          },
        },
      ],
      "kind": "Document",
    },
    Object {
      "definitions": Array [
        Object {
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "GeneratedClientQuery",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "__typename",
                },
              },
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "authToken",
                },
              },
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "mediaPlayer",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "__typename",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "currentTrack",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isPlaying",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isFullscreen",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isVisible",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "currentTime",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "showVideo",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "muted",
                      },
                    },
                  ],
                },
              },
            ],
          },
        },
      ],
      "kind": "Document",
    } => Object {
      "definitions": Array [
        Object {
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "GeneratedClientQuery",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "__typename",
                },
              },
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "authToken",
                },
              },
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "mediaPlayer",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "__typename",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "currentTrack",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isPlaying",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isFullscreen",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isVisible",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "currentTime",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "showVideo",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "muted",
                      },
                    },
                  ],
                },
              },
            ],
          },
        },
      ],
      "kind": "Document",
    },
    Object {
      "definitions": Array [
        Object {
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "GeneratedClientQuery",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "__typename",
                },
              },
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "authToken",
                },
              },
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "mediaPlayer",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "__typename",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "currentTrack",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isPlaying",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isFullscreen",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isVisible",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "currentTime",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "showVideo",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "muted",
                      },
                    },
                  ],
                },
              },
            ],
          },
        },
      ],
      "kind": "Document",
    } => Object {
      "definitions": Array [
        Object {
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "GeneratedClientQuery",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "__typename",
                },
              },
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "authToken",
                },
              },
              Object {
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "mediaPlayer",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "__typename",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "currentTrack",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isPlaying",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isFullscreen",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isVisible",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "currentTime",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "showVideo",
                      },
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "muted",
                      },
                    },
                  ],
                },
              },
            ],
          },
        },
      ],
      "kind": "Document",
    },
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "track",
          },
          "operation": "mutation",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "input",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "input",
                      },
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "trackEvent",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "success",
                      },
                      "selectionSet": undefined,
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "defaultValue": undefined,
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "AnalyticsTrackInput",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "input",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 106,
        "start": 0,
      },
    } => Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "track",
          },
          "operation": "mutation",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "input",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "input",
                      },
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "trackEvent",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "success",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "__typename",
                      },
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "defaultValue": undefined,
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "AnalyticsTrackInput",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "input",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 106,
        "start": 0,
      },
    },
  },
  "watches": Set {},
  "write": [Function],
}
`;
